src/Tools/AnalyzerRunner/Program.cs

```diff
@@ -0,0 +1,491 @@
+                        if (documentCount > 0)
+                        {
+                            projectPerformance[project.Id] = sumOfDocumentAverages / documentCount;
+                        }
+                    }
+
+                    var slowestFiles = documentPerformance.OrderBy(pair => pair.Value.EditsPerSecond).GroupBy(pair => pair.Key.ProjectId);
+                    Console.WriteLine("Slowest files in each project:");
+                    foreach (var projectGroup in slowestFiles)
+                    {
+                        Console.WriteLine($"  {solution.GetProject(projectGroup.Key).Name}");
+                        foreach (var pair in projectGroup.Take(5))
+                        {
+                            var document = solution.GetDocument(pair.Key);
+                            Console.WriteLine($"    {document.FilePath ?? document.Name}: {pair.Value.EditsPerSecond:0.00}");
+                        }
+                    }
+
+                    foreach (var projectId in solution.ProjectIds)
+                    {
+                        if (!projectPerformance.TryGetValue(projectId, out var averageEditsInProject))
+                        {
+                            continue;
+                        }
+
+                        var project = solution.GetProject(projectId);
+                        Console.WriteLine($"{project.Name} ({project.DocumentIds.Count} documents): {averageEditsInProject:0.00} edits per second");
+                    }
+                }
+
+                foreach (var group in allDiagnostics.GroupBy(diagnostic => diagnostic.Id).OrderBy(diagnosticGroup => diagnosticGroup.Key, StringComparer.OrdinalIgnoreCase))
+                {
+                    Console.WriteLine($"  {group.Key}: {group.Count()} instances");
+
+                    // Print out analyzer diagnostics like AD0001 for analyzer exceptions
+                    if (group.Key.StartsWith("AD", StringComparison.Ordinal))
+                    {
+                        foreach (var item in group)
+                        {
+                            Console.WriteLine(item);
+                        }
+                    }
+                }
+
+                if (!string.IsNullOrWhiteSpace(options.LogFileName))
+                {
+                    WriteDiagnosticResults(analysisResult.SelectMany(pair => pair.Value.GetAllDiagnostics().Select(j => Tuple.Create(pair.Key, j))).ToImmutableArray(), options.LogFileName);
+                }
+            }
+        }
+
+        private static async Task<DocumentAnalyzerPerformance> TestDocumentPerformanceAsync(ImmutableArray<DiagnosticAnalyzer> analyzers, Project project, DocumentId documentId, Options analyzerOptionsInternal, CancellationToken cancellationToken)
+        {
+            // update the project compilation options
+            var modifiedSpecificDiagnosticOptions = project.CompilationOptions.SpecificDiagnosticOptions.Add("AD0001", ReportDiagnostic.Error);
```

Comment:
Also report errors for [AD0002](http://source.roslyn.io/#Microsoft.CodeAnalysis/DiagnosticAnalyzer/AnalyzerExecutor.cs,31) when the driver faults.